### PR TITLE
Docs: Clarify --rule description (fixes #2773)

### LINKED ISSUE #2773: Docs: --rule also needs --no-eslintrc to "start fresh"
I am adding ESLint to an existing project and have many failures to clean up. I would like to use the CLI to apply only one rule at a time, fix only those issues, and then commit just those fixes, so that I have a single rule per commit.  However, I am finding this difficult.

My plan was to use `eslint --reset --rule 'quotes: [2, single]' lib` so that only the quotes rule is applied, and none of the default rules, as suggested in the docs:

> **--rule**
This option specifies rules to be used. They will be merged into any previously defined rules. To start fresh, simply combine with the --reset flag. 

But I am extending [standard](https://github.com/feross/eslint-config-standard) in my .eslintrc and the `--reset` flag does not reset rules in plugins (as mentioned in https://github.com/eslint/eslint/issues/2100#issuecomment-103728896).  

I am also using ECMA modules (`"ecmaFeatures": {"modules": true}` which I cannot find a way to set from the CLI, so using `--no-eslintrc` isn't an option to avoid extending standard.  For the time being I can remove the plugin from my .eslintrc, but it seems like maybe there should be a better way?

Edit: Even when using `--reset`, the rules defined in my `.eslintrc` are still applied.  In my opinion, the documentation for `--rule` is a little misleading.  To really "start fresh", it seems necessary to use `--no-eslintrc` in addition to `--reset` (at least until v1.0).

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

File: docs/user-guide/command-line-interface.md [hunk 1]
```diff
@@ -182,7 +182,8 @@ Example:
 
 ### `--rule`
 
-This option specifies rules to be used. They will be merged into any previously defined rules. To start fresh, simply combine with the `--reset` flag. To define multiple rules, separate them using commas, or use the flag multiple times. The [levn](https://github.com/gkz/levn#levn--) format is used for specifying the rules.
+This option specifies rules to be used. These rules will be merged with the default rules and any rules specified with configuration files. (You can use `--reset` and `--no-eslintrc`, respectfully, to change those behaviors.) To define multiple rules, separate them using commas, or use the flag multiple times. The [levn](https://github.com/gkz/levn#levn--) format is used for specifying the rules.
```
Comment (btmills) [inline docs/user-guide/command-line-interface.md]: Looks like "respectively" got autocorrected to "respectfully."

Comment (nzakas) [inline docs/user-guide/command-line-interface.md]: Doh! I hate being stuck on this tablet!

**Comment by nzakas** [comment]
Crap, can't squash commits through github, so I can't edit :(

**Comment by btmills** [comment]
"Reopened" as #2776. (Apparently PRs from branches that have been force-pushed since closing can't be reopened.)

**Pull request closed without merging**